static/templyte/src/app/download-template/download-template.component.ts

```diff
@@ -68,6 +73,45 @@ export class DownloadTemplateComponent implements OnInit {
         this.authService.logout(null);
     }
 
+    storeRating(new_rating) {
+        const component = this;
+        const authorUID:string  =  component.user.uid;
+        component.templateRatingsInfoDatabaseRef.once('value').then(snapshot2 => {
+            const old_rating = snapshot2.val(); //value of previous rating
+            var varNumRatings = 0;
+            var varRatingSum = 0;
+            component.templateRatingsInfoDatabaseRef.set(
+                 new_rating
+            );
+            let hasRated:boolean = old_rating != null;
+            if (hasRated) {
+                component.templateDirectoryInfoDatabaseRef.child('/ratingSum').transaction(function(ratingSum){
+                    varRatingSum = ratingSum - old_rating + new_rating;
+                    return ratingSum - old_rating + new_rating;
+                }).then(function(ratingSum) {
+                    varNumRatings = 1;
+                    component.templateDirectoryInfoDatabaseRef.child('/averageRating').transaction(function(avgRating){
+                            return (varRatingSum * 1.0) / varNumRatings;
+                    });
+                });
+            } else {
+                component.templateDirectoryInfoDatabaseRef.child('/numberRatings').transaction(function(numberRatings){
+                    varNumRatings = numberRatings + 1;
+                    return numberRatings + 1;
+                }).then(function(ratingSum) {
+                    component.templateDirectoryInfoDatabaseRef.child('/ratingSum').transaction(function(ratingSum){
+                        varRatingSum = ratingSum + new_rating;
+                        return ratingSum + new_rating;
+                    }).then(function(ratingSum) {
```

Comment:
Shadowed name - dangerous

Reply:
Are you using this? - it's probably safer to use the result of the transaction to make changes.